WordPress/gutenberg
# [Edit Post]: Hide Template part in post editor

Resolves: https://github.com/WordPress/gutenberg/issues/30668

This PR takes an approach of introducing a new `__experimentalDisallowedBlocks` property which is used in `Post Content` block, in combination of hiding the Template Part block when we are not in template mode in post editor.

We need both because in post editor the post content is not rendered as a single `Post Content` block.

By not unregistering the block the existing content can work as before without disruptions. 

## Testing instructions
1. Go to a post/page and try to find through the inserter the  `Template Part`
2. Observe that you can't
3. Edit the `post/page template` and observe that the block is available only outside the `Post Content` block
4. Exit the template editing and observe that it's again unavailable in post editor
5. Observe that existing Template Parts in a post/page work properly as before


## Notes
I'll update the tests if the approach is deemed okay.
